tests/test_ERst.cpp

```diff
@@ -221,3 +226,659 @@ BOOST_AUTO_TEST_CASE(TestERst_3) {
     };
 }
 
+// ====================================================================
+
+class RSet
+{
+public:
+    explicit RSet(std::string base)
+        : odir_(boost::filesystem::temp_directory_path() /
+                boost::filesystem::unique_path("rset-%%%%"))
+        , base_(std::move(base))
+    {
+        boost::filesystem::create_directories(this->odir_);
+    }
+
+    ~RSet()
+    {
+        boost::filesystem::remove_all(this->odir_);
+    }
+
+    operator ::Opm::EclIO::OutputStream::ResultSet() const
+    {
+        return { this->odir_.string(), this->base_ };
+    }
+
+private:
+    boost::filesystem::path odir_;
+    std::string             base_;
+};
+
+namespace {
+    template <class Coll>
+    void check_is_close(const Coll& c1, const Coll& c2)
+    {
+        using ElmType = typename std::remove_cv<
+            typename std::remove_reference<
+                typename std::iterator_traits<
+                    decltype(std::begin(c1))
+                >::value_type
```

Comment:
You do have a funcy notation.

Reply:
It's a compromise.  The declaration doesn't fit on a line and has a hierarchical structure so I thought I'd underscore that fact.